test/elements.js

```diff
@@ -104,6 +104,53 @@ test('adjacent text nodes', function (t) {
   t.end()
 })
 
+test('space in only-child text nodes', function (t) {
+  t.plan(1)
+  var result = bel`
+    <span>
+      surrounding
+      newlines
+    </span>
+  `
+  t.equal(result.outerHTML, '<span>surrounding\n      newlines</span>', 'should remove extra space')
```

Comment:
An earlier version of this patch collapsed space within text nodes, as well.

Other than the negligible RAM for the space characters, is there any real perf benefit to messing with text nodes that will definitely end up in the rendered tree?